src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,283 @@
+        /// </summary>
+        public string Title
+        {
+            get { return this.GetParameterValue<string>("title"); }
+        }
+
+        /// <summary>
+        /// The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.
+        /// </summary>
+        public MediaRange Type
+        {
+            get { return this.GetParameterValue<MediaRange>("type"); }
+        }
+
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>
+        /// true if the current object is equal to the <paramref name="other" /> parameter; otherwise, false.
+        /// </returns>
+        public bool Equals(HttpLink other)
+        {
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            if (this.TargetUri != other.TargetUri)
+            {
+                return false;
+            }
+
+            foreach (var parameter in other.Parameters)
+            {
+                object parameterValue;
+                if (!this.parameters.TryGetValue(parameter.Key, out parameterValue))
+                {
+                    return false;
+                }
+
+                if (parameterValue == null || parameter.Value == null)
+                {
+                    continue;
+                }
+
+                if (!parameterValue.Equals(parameter.Value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object" />, is equal to this instance.
+        /// </summary>
+        /// <param name="other">The <see cref="System.Object" /> to compare with this instance.</param>
+        /// <returns>
+        ///   <c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.
+        /// </returns>
+        public override bool Equals(object other)
+        {
+            return ((IEquatable<HttpLink>)this).Equals(other as HttpLink);
+        }
+
+        /// <summary>
+        /// Returns a hash code for this instance.
+        /// </summary>
+        /// <returns>
+        /// A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.
+        /// </returns>
+        public override int GetHashCode()
+        {
+            return StringComparer.OrdinalIgnoreCase.GetHashCode(this.ToString());

```

Comment:
Just something to think about. Do we really want this to generate a potentially big string each time it needs the hash code? I think that might be breaking one of the "rules" of hash code functions...